scripts/addValidCourses.sql

```diff
@@ -0,0 +1,29 @@
+CREATE FUNCTION public."ValidCourses"(IN "P_person_id" integer)
```

Comment:
Despite this being a super-efficient way of getting all courses an individual person is qualified for by relying on the database queries, it would be inefficient to use this Function repeatedly by the Java lambda as the same database entry will be returned multiple times.

The solution is to either:
1. Use the Java "as is"
2. Build a View that gets every person_id like this Function.
    - This is easy to do, but the gains are not likely to be very much, at least until the Java is proven to be slow.